src/win/core.c

```diff
@@ -442,3 +443,32 @@ int uv__socket_sockopt(uv_handle_t* handle, int optname, int* value) {
 
   return 0;
 }
+
+
+int uv_homedir(char* home, size_t* len) {
+  wchar_t* path;
+  HRESULT hr;
+  int r;
+
+  if (home == NULL)
+    return -EINVAL;
+
+  hr = SHGetKnownFolderPath(FOLDERID_Profile,
+                            0,
+                            NULL,
+                            &path);
+
+  if (hr != S_OK)
+    return uv_translate_sys_error(WSAGetLastError());
+
+  r = wcstombs(home, path, len);
+  CoTaskMemFree(path);
+
+  if (r < 0)
+    return uv_translate_sys_error(WSAGetLastError());
+
+  home[r] = '\0';
+  *len = r;
```

Comment:
What's implied here is that the returned length *doesn't* include the null terminator.
I have no particular opinion on whether that's good or bad, but please ensure that this API behaves the same as the unix version, and be consistent with other path-returning functions (e.g. `uv_cwd()`)